Lib9c/Action/Guild/ApplyGuild.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using Bencodex.Types;
+using Libplanet.Action;
+using Libplanet.Action.State;
+using Nekoyume.Extensions;
+using Nekoyume.Module.Guild;
+using Nekoyume.TypedAddress;
+
+namespace Nekoyume.Action.Guild
+{
+    [ActionType(TypeIdentifier)]
+    public class ApplyGuild : ActionBase
+    {
+        public const string TypeIdentifier = "apply_guild";
+
+        private const string GuildAddressKey = "ga";
+
+        public ApplyGuild() {}
+
+        public ApplyGuild(GuildAddress guildAddress)
+        {
+            GuildAddress = guildAddress;
+        }
+
+        public GuildAddress GuildAddress { get; private set; }
+
+        public override IValue PlainValue => Dictionary.Empty
+            .Add("type_id", TypeIdentifier)
+            .Add("values", Dictionary.Empty
+                .Add(GuildAddressKey, GuildAddress.Bencoded));
+
+        public override void LoadPlainValue(IValue plainValue)
+        {
+            var root = (Dictionary)plainValue;
+            if (plainValue is not Dictionary ||
+                !root.TryGetValue((Text)"values", out var rawValues) ||
+                rawValues is not Dictionary values ||
+                !values.TryGetValue((Text)GuildAddressKey, out var rawGuildAddress))
+            {
+                throw new InvalidCastException();
+            }
+
+            GuildAddress = new GuildAddress(rawGuildAddress);
+        }
+
+        public override IWorld Execute(IActionContext context)
+        {
+            context.UseGas(1);
+            var world = context.PreviousState;
+            var signer = context.GetAgentAddress();
+
+            if (world.GetJoinedGuild(signer) is not null)
+            {
+                throw new InvalidOperationException("The signer is already joined in a guild.");
+            }
+
+            // NOTE: Check there is such guild.
+            _ = world.GetGuild(GuildAddress);
```

Comment:
Wouldn't it be better to use TryGetGuild for consistency?

Reply:
I fixed it in [5cd8026](https://github.com/planetarium/lib9c/pull/2719/commits/5cd8026b64d9c0cf803e848666cb3af92adf3302)